beacon-chain/blockchain/receive_block.go

```diff
@@ -149,27 +148,16 @@ func (s *Service) ReceiveBlockBatch(ctx context.Context, blocks []*ethpb.SignedB
 		// Reports on blockCopy and fork choice metrics.
 		reportSlotMetrics(blockCopy.Block.Slot, s.headSlot(), s.CurrentSlot(), s.finalizedCheckpt)
 	}
+
 	lastBlk := blocks[len(blocks)-1]
 	lastRoot := blkRoots[len(blkRoots)-1]
-
-	if err := s.stateGen.SaveState(ctx, lastRoot, postState); err != nil {
```

Comment:
I dont see how this works, we need to save the post state so that we can process the next batch. Otherwise state gen will have to regenerate the state when processing the next batch.

Reply:
Ah i see. I thought saving it per boundary would be enough. I'll add this back

Reply:
Added it here. https://github.com/prysmaticlabs/prysm/pull/6599/commits/56ca04d96065533d7ebb901318d1a4e850a931bc

Going to test this now